ovirt_imageio/tool/__init__.py

```diff
@@ -0,0 +1,101 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Tool for transferring disk images.
+"""
+
+import argparse
+import getpass
+
+from contextlib import closing
+
+from .. import client
+from . import ovirt
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Transfer disk images")
+    parser.set_defaults(
+        command=lambda _: parser.print_help())
+
+    commands = parser.add_subparsers(title="commands")
+
+    cmd = commands.add_parser("download-disk", help="Download disk")
+    cmd.set_defaults(command=download_disk)
+
+    add_common_options(cmd)
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Download image format (default qcow2).")
+
+    cmd.add_argument(
+        "disk_id",
+        help="Disk ID to download.")
+
+    cmd.add_argument(
+        "filename",
+        help="Target filename.")
+
+    args = parser.parse_args()
+
+    # XXX Configure logging
+
+    fill_password(args)
+
+    args.command(args)
+
+
+def add_common_options(cmd):
+    cmd.add_argument(
+        "--engine-url",
+        help="ovirt-engine URL.")
+
+    cmd.add_argument(
+        "--username",
+        help="ovirt-engine username.")
+
+    cmd.add_argument(
+        "--password-file",
+        help="Read ovirt-engine password from file.")
+
+    cmd.add_argument(
+        "--cafile",
+        help="Path to ovirt-engine CA certificate")
+
+
+def fill_password(args):
+    if args.password_file:
+        with open(args.password_file) as f:
+            args.password = f.read().rstrip("\n")
```

Comment:
Password as plain text is good enough here?

Reply:
How your private key is protected? This password file can be protected in the way.

If users do not want to keep the password in a file, they can avoid --password-file. In this case the tool ask the password on every invocation using getpass().

An additional way we be using the keyring - but it is harder to implement and get work on multiple desktop environment.

When running in a k8s, the password will be provided as a secret, available only inside a pod.

Reply:
If we don't want to use a hash algorithm to store the password (which may require to depend on a non-standard library, and more work than is worth it), we could also read environment variables where the password is stored.

That is a good mid-way strategy between prompting the user, which may be annoying specially for consecutive calls, and having the raw password stored in a file.

Reply:
We cannot hash the password, we need the plaintext for authentication. We can enctypt the password, but then you need the key to decrypt it.